src/sentry/sentry_metrics/querying/data_v2/execution.py

```diff
@@ -348,16 +304,15 @@ def _align_date_range(cls, metrics_query: MetricsQuery) -> tuple[MetricsQuery, i
         # We use as a reference the interval supplied via the initial version of the query.
         interval = metrics_query.rollup.interval
         if interval:
-            modified_start, modified_end, intervals_number = to_intervals(
+            modified_start, modified_end, intervals_number = adjust_time_bounds_with_interval(
```

Comment:
Moved to an internal impl of this function since I wanted to remove the only dependency to the old layer.